contracts/parent-chain/contract-upgrades/EspressoOspMigrationAction.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity ^0.8.9;
+
+import "forge-std/Script.sol";
+import "@arbitrum/nitro-contracts/src/challenge/IChallengeManager.sol";
+import "@arbitrum/nitro-contracts/src/rollup/IRollupCore.sol";
+import "@arbitrum/nitro-contracts/src/rollup/IRollupAdmin.sol";
+
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+error IncorrectWasmModuleRoot(bytes32 incorrectAddr);
+
+error AddressIsNotContract(address incorrectAddr);
+
+error ChallengeManagerUpdated(address newChallengeManagerAddr);
+
+error OspNotUpgraded(address oldOspAddress);
+
+error WasmModuleRootNotUpdated(bytes32 oldWasmModuleRoot);
+
+
+
+contract OspMigrationAction{ 
+    address public immutable newOspEntry;
+    bytes32 public immutable newWasmModuleRoot;
+    bytes32 public immutable currentWasmModuleRoot;
+    address public immutable currentOspEntry;
+    address public immutable rollup;
+    address public immutable proxyAdmin;
+
+
+    constructor(
+        address _newOspEntry,
+        bytes32 _newWasmModuleRoot,
+        address _currentOspEntry,
+        bytes32 _currentWasmModuleRoot,
+        address _rollup,
+        address _proxyAdmin
+    ){
+        newOspEntry = _newOspEntry;
```

Comment:
perhaps add some error checking for these variables in the constructor

Reply:
If we add the checks in the perform function to the constructor we can, in theory, remove them from the perform function to avoid using extra gas correct?